<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html>

<head>
<title>DNA Data Visualization (DDV) software - by Tomasz Neugebauer</title>

<link rel="stylesheet" type="text/css" href="../seadragon.css" />
<style type="text/css">
body {padding:20px 4% 200px 2%;}
</style>
</head><body>
<h1>DNA Data Visualization (DDV) software - Download/Install/Help</h1>
<h2 class="mainTitle"><a href="../index.html">About &gt;</a> </h2>
<h3>Summary:</h3>

<p>This free open source C# application creates visualizations of FASTA formatted DNA nucleotide data.  
  DDV can download FASTA sequences from NIH, or use local FASTA formatted files.  
  DDV generates interfaces based on OpenSeadragon multiscale visualizations of DNA  combined with BioJS  data browsing. </p>
<p>The interfaces generated by DDV are constructed with multiscale images, HTML, JavaScript, and PHP. Generating interfaces with DDV requires a Windows operating system (7 or higher). The interfaces generated with DDV can be used on any web browser and operating system.</p>
<h3>Download:</h3>
<p>
<a href="https://github.com/photomedia/DDV/raw/master/bin/ZIP/DDV.zip">DDV.zip</a> - Release v1.1 ZIP file at Github<br />
Contains compiled binaries (1.0.6086.32371) for Windows (7 and higher)
</p>

<h3>Installation/use:</h3>
<p>DDV generator is intended to run on Windows OS (7 and higher).<br />
Download and unzip the binaries into a local installation folder, for example: C:/DATA/DDV/<br />
DDV requires .NET Framework 4.0 or higher to be installed (see System Requirements below) <br />
Double-click on "DDV.exe" to launch the application <br />
DDV generates output in the /output/ folder under the installation directory, for example: C:/DATA/DDV/output/ 
</p>

<h3>DDV Source Code:</h3>
<p>
The source code for this application is available at <a href="https://github.com/photomedia/DDV/">Github</a>.
</p>


<h3>System Requirements:</h3>
<p>
Running DDV generator requires Windows OS (7 and higher)
<br />
.NET Framework 4.0 or higher is required.  This is included by default in the above versions of Windows.  
If you need to install the .NET Framework because your version of Windows does not have it for some reason, the installer is found here:
<ul>
<li>
<a href="http://www.microsoft.com/en-ca/download/details.aspx?id=17851">.NET Framework 4</a> or higher.  
</li>
</ul>

  Accessing visualizations generated with DDV require a web browser with JavaScript.
<p>
Note: The DDV zip file includes the msvcr110.dll needed by Civetweb.  You may also choose to install it instead with the <a href="https://www.microsoft.com/en-ca/download/details.aspx?id=30679">Visual C++ Redistributable</a> (x86 version, vcredist_x86). 
</p>
</p>



<h3 id="help">Help:</h3>
<h4>Generating visualizations with DDV</h4>
<p>Follow the instructions for Installation/use. When you open DDV.exe, you will see the following interface:</p>
<p><img src="DDV-main.jpg" alt="Windows DDV interface" width="1151" height="692" /></p>
<p>The workflow for generating visualizations consists of 4 steps, activated in order by clicking on the buttons indicated:</p>
<ul>
  <li>First step: Input NCBI Accession.Version or GI (1a), and download FASTA file using NCBI eFetch (1), or select a local FASTA file (1b).</li>
  <li>Second step: Read Sequence Properties (2). This reads the name, length and  sequence identifiers such as Accession.Version and/or GI number and displays these in the &quot;Sequence Properties&quot; panel.</li>
  <li>Third step: If necessary, modify   image width (3a) and then click on &quot;Generate Image and Interface&quot;
(3) This initializes an image of appropriate  height for the  visualizations. This parameter depends on the sequence size - the smaller the height, the larger will be the width.  For relatively small genomes (eg: bacteria), a height of 3000 is recommended. For large data sets (eg.: homo sapiens), a height of 20000 is recommended. This step processes all of the nucleotides in the sequence by  painting colors for each nucleotide on the image, generating a master image of  large size. </li>
  <li>Fourth step: If necessary, modify tile size (4a) and then click on &quot;Process Image with Deep Zoom&quot; (4). This generates the Deep Zoom Images (DZI) pyramid and XML  from the master image using DeepZoomTools.dll. It also generates the HTML, CSS and JavaScript files for the  completed DNA Data Visualization interface and places the results in the /output/ folder. </li>
</ul>
<p>The results are placed in the [DDV]<strong>/output/</strong> folder. The files in this /output/ folder can also be uploaded to a web server that supports PHP for sharing the generated visualizations with others. You can click on &quot;Browse Output Folder&quot; (6) button to view the generated files with Windows.</p>
<p>The generated interface should open automatically in your browser, with the address &quot;http://localhost:1818/...&quot; You can also click on &quot;Open Generated Interfaces&quot; (5) to view  the list of generated visualizations and navigate.  Please note that when navigating the visualizations locally, without uploading to a web server, DDV.exe application has to remain open in order for the visualizations to function fully. This is because the DDV application is  running Civetweb (a minimal web server) to create the same environment as a PHP-supporting web server. Shutting down DDV also closes Civetweb.</p>
<h4>Navigating/exploring generated visualizations</h4>
<p>A generated visualization looks like the following:</p>
<p><img src="viz-interface.jpg" longdesc="example visualized DNA data interface using DDV" /></p>
<p>Navigating the DDV visualizations is accomplished with a combination of mouse (or touchscreen) and keyboard.  Zooming in and out is done with the mouse wheel or by clicking on the +/- navigation buttons on the image (1). In addition, there is a button for full-screen (4) and another for &quot;home&quot;. The following keys on the keyboard are used to select the beginning and end of a sequence fragment selected for computation (%GC and GC Skew)::</p>
<ul>
  <li><strong>[ q ]</strong>- mark beginning of sequence fragment</li>
  <li><strong>[ z ] </strong>- mark end of sequence fragment</li>
</ul>
<p>Navigating around the image can be done with the visual navigator in the top right (2), by dragging with the mouse, and/or with the following keys on the keyboard:</p>
<ul>
  <li><strong>[ w, up arrow ]</strong> - move viewport <strong>up</strong></li>
  <li><strong>[ s, down arrow ]</strong> - move viewport <strong>down</strong></li>
  <li><strong>[ a, left arrow ]</strong> - move viewport <strong>left</strong></li>
  <li><strong>[ d, right arrow ]</strong> - move viewport <strong>right</strong></li>
  <li><strong>[ 0 ]</strong> - zoom / move viewport <strong>home</strong></li>
  <li><strong>[ - / _, shift+W, shift+up arrow ]</strong> - zoom viewport <strong>out</strong></li>
  <li><strong>[ = / +, shift+S, shift+down arrow ]</strong> - zoom viewport <strong>in</strong></li>
</ul>
<h5>Relative scale bar display</h5>
<p>The  generated visualizations also leverage OpenSeadragon’s scale bar (3).  The bar is customized so that it shows the  relationship between width on the image and the number of nucleotides, as the  user zooms in and out. </p>
<h5>%G+C nucleic composition computation</h5>
<p>The visualization uses a combination of JavaScript and PHP to compute %G+C and  nucleic composition density of a selected portion  of the visualized sequence, and display the computed results on the  interface. Clicking on &quot;Compute A, T, G, C, G+C, A+T&quot; (5) sends the results for the selected sequence (based on the &quot;start&quot; and &quot;end&quot; positions set) to the Result log (5a).</p>
<h5>GC Skew Graph</h5>
<p>The visualization uses JavaScript  for  computing  GC Skew data and plotting it on the interface using D3.js library.  The skew window is set depending on the  number of nucleotides in the sequence, ranging from 50 for sequences of less  than 100000 nucleotides to 10000 for sequences longer than 10000000 nucleotides. Clicking on &quot;Generate GC Skew&quot; (6) computes the GC Skew graph (based on the &quot;start&quot; and &quot;end&quot; positions set) for the sequence portion selected and places the visualization (6a) in the Result log. </p>
<h5>Partial sequence data display and copy:</h5>
<p>In addition, as you point on the data with the mouse pointer, the currently pointed at nucleotide and its surrounding 210 nucleotides are displayed to the right of the multiscale image (7). The following key is used to copy that fragment to result log (7a):</p>
<ul>
  <li> <strong>[ x ]</strong> - copy 210 bp fragment to result log</li>
</ul>
<p> Click on &quot;x&quot; to copy the currently pointed at sequence fragment to the Result log. This is implemented with the  help of BioJS.  Please note that this feature is activated only after the sequence is loaded  - before the loading is complete, the interface displays information regarding progress. </p>
<div></div>
<h3>FAQ:</h3>
<ul>
  <li>Q: What  components are a part of DDV?</li>
  <li>A: The following are a part of DDV: jQuery, BioJS.Sequence, D3.js, PHP, OpenSeadragon, DeepZoomTools.dll, Civetweb. </li>
</ul>
<ul>
  <li>Q: Why does DDV require FASTA sequences with 70 nucleotides per line?</li>
  <li>A: FASTA format specifies that each line is to have  80 nucletoides or less. 70 nucleotides per line is the most common length that we found. NCBI eFetch API, which is used by DDV to retrieve sequences based on GI, consistently returns sequences with  length 70. The source code of DDV leaves the line length  a variable in its scripts (iLineLength), leaving the potential to accomodate other lengths. By default in this version of DDV, however, if you wish to visualize a FASTA file with line length other than 70 nucleotides using DDV, you will have to pre-process the FASTA file, converting it to line length 70.</li>
</ul>
<ul>
  
  <li>Q: Why are the visualization results served from localhost:1818?</li>
  <li>A: DDV generates visualizations that can be viewed locally and placed on a web server.  
  	The interaction with the FASTA dataset file, and PHP scripts require a "localhost" environment, provided for with Civetweb and PHP. It is possible to change the default DDV port from 1818 to another, by modifying the civetweb.conf file found in the DDV installation folder. To do this, modify the line &quot;listening_ports 127.0.0.1:1818&quot;.</li>
</ul>
<ul>
  <li>Q: Where can I send my comments/suggestions regarding DDV?</li>
  <li>A: Please send your comments/suggestions to tomasz.neugebauer@concordia.ca, please include  &quot;DDV&quot; in the subject of the email.</li>
</ul>
<h3>Credits</h3>
<p>
The software was created by Tomasz Neugebauer (tomasz.neugebauer@concordia.ca), Concordia University Libraries, Concordia University.
Collaboration for the biological aspects of the project: Éric Bordeleau; Vincent Burrus and Ryszard Brzezinski; Département de biologie; Faculté des Sciences; Université de Sherbrooke.</p>




</body>

</html>
